### PR TITLE
Integrate starter pack

### PR DESCRIPTION
**Pull request by motechFR**

### CODE REVIEW

File: packages/scoutgame/src/builderNfts/artwork/utils.ts [hunk 1]
```diff
@@ -7,14 +7,22 @@ export function getNftTokenUrlPath({
...
+export function getNftCongratsPath({
+  season,
+  tokenId,
+  starterPack
+}: {
+  season: string;
+  tokenId: number;
+  starterPack?: boolean;
+}) {
+  return `seasons/${season}/${contractName}/${tokenId}/${starterPack ? 'starter-pack-' : ''}${'congrats.png'}`;
```
Comment (mattcasey) [inline packages/scoutgame/src/builderNfts/artwork/utils.ts:25]: we should be using a different contractName for starter-pack, and then you do not need a special filename.

File: scripts/optimism/proposalReviewerStats/stubs.ts
Comment (mattcasey) [inline scripts/optimism/proposalReviewerStats/stubs.ts:1]: this should be a separate pr

File: public/mockServiceWorker.js [hunk 2]
```diff
@@ -8,123 +8,123 @@
  * - Please do NOT serve this file on production.
  */
 
-const PACKAGE_VERSION = '2.2.14';
-const INTEGRITY_CHECKSUM = '26357c79639bfa20d64c0efca2a87423';
-const IS_MOCKED_RESPONSE = Symbol('isMockedResponse');
-const activeClientIds = new Set();
+const PACKAGE_VERSION = '2.2.14'
```
Comment (mattcasey) [inline public/mockServiceWorker.js:11]: not sure if this file was intended to change. it probably should have semicolons?

File: packages/scoutgame/src/points/dividePointsBetweenBuilderAndScouts.ts [hunk 3]
```diff
@@ -31,14 +31,26 @@ export async function dividePointsBetweenBuilderAndScouts({
...
     }
   });
 
   const { totalNftsPurchased, nftsByScout } = nftPurchaseEvents.reduce(
     (acc, purchaseEvent) => {
-      acc.totalNftsPurchased += purchaseEvent.tokensPurchased;
-      acc.nftsByScout[purchaseEvent.scoutId] =
-        (acc.nftsByScout[purchaseEvent.scoutId] || 0) + purchaseEvent.tokensPurchased;
+      // Normal NFTs are 10x more valuable than Starter Pack NFTs
+      const multiplier = purchaseEvent.builderNFT.nftType === 'season_1_starter_pack' ? 1 : 10;
```
Comment (mattcasey) [inline packages/scoutgame/src/points/dividePointsBetweenBuilderAndScouts.ts:49]: this would be nice to define in a constants someplace

File: packages/scoutgame/src/builderNfts/constants.ts [hunk 4]
```diff
@@ -30,6 +31,15 @@ export function getBuilderContractAddress(): `0x${string}` {
   return (env('BUILDER_NFT_CONTRACT_ADDRESS') || process.env.REACT_APP_BUILDER_NFT_CONTRACT_ADDRESS) as `0x${string}`;
 }
 
+export function getBuilderStarterPackContractAddress(): `0x${string}` {
```
Comment (mattcasey) [inline packages/scoutgame/src/builderNfts/constants.ts:34]: you should not need to wrap this in a function

File: packages/scoutgame/src/builderNfts/constants.ts [hunk 5]
```diff
@@ -30,6 +31,15 @@ export function getBuilderContractAddress(): `0x${string}` {
...
+  return (env('BUILDER_STARTER_PACK_CONTRACT_ADDRESS') ||
+    process.env.REACT_APP_BUILDER_STARTER_PACK_CONTRACT_ADDRESS) as `0x${string}`;
+}
+
+export function getBuilderContractAddressForNftType(nftType: BuilderNftType): `0x${string}` {
```
Comment (mattcasey) [inline packages/scoutgame/src/builderNfts/constants.ts:39]: also should not be a function

File: packages/scoutgame/src/builders/approveBuilderStarterPack.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    where: {
+      id: builderId
+    },
+    data: {
+      builderStatus: 'approved'
+    }
+  });
+
+  // do not wait for git to complete
+  importReposByUser(scout.githubUser[0]?.login).catch((error) => {
```
Comment (mattcasey) [inline packages/scoutgame/src/builders/approveBuilderStarterPack.ts:45]: do not import repos automatically

File: apps/scoutgamecron/src/tasks/resolveMissingPurchases/index.ts [hunk 7]
```diff
@@ -1,11 +1,12 @@
...
 import type Koa from 'koa';
 
-import { findAndIndexMissingPurchases } from './findAndIndexMissingPurchases';
+import { findAndIndexMissingPurchases, findAndIndexMissingStarterPackPurchases } from './findAndIndexMissingPurchases';
 
 export async function resolveMissingPurchasesTask(ctx: Koa.Context) {
   scoutgameMintsLogger.info('Resyncing builder NFT sales');
 
   await findAndIndexMissingPurchases();
+  await findAndIndexMissingStarterPackPurchases();
```
Comment (mattcasey) [inline apps/scoutgamecron/src/tasks/resolveMissingPurchases/index.ts:10]: would be nicer if you could just call the same method but with different contract addresses or something

File: apps/scoutgame/components/scout/ScoutPage.tsx [hunk 8]
```diff
@@ -66,6 +70,14 @@ export function ScoutPage({
             gap: 2
           }}
         >
+          {starterpackBuilders && (
```
Comment (mattcasey) [inline apps/scoutgame/components/scout/ScoutPage.tsx:73]: we should check that the list is empty too, right?

File: apps/scoutgame/app/(info)/welcome/scout-info/page.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+    robots: 'noindex'
+  },
+  title: 'Welcome'
+};
+
+export default async function ScoutInfo() {
+  // FID of the builder that we want to show in the ScoutInfoPage
+  const builderfId = 547807;
+  const [, builderByFid] = await safeAwaitSSRData(
+    getBuildersByFid({ fids: [builderfId], limit: 1, season: currentSeason })
```
Comment (mattcasey) [inline apps/scoutgame/app/(info)/welcome/scout-info/page.tsx:21]: this must not block the page component

File: apps/scoutgame/scripts/test.ts
Comment (mattcasey) [inline apps/scoutgame/scripts/test.ts:1]: this filename doesnt mean anything. use 'query.ts' for random scripts

File: packages/scoutgame/src/builders/getStarterpackBuilders.ts [hunk 10]
```diff
@@ -0,0 +1,71 @@
+import { BuilderNftType, prisma } from '@charmverse/core/prisma-client';
+
+import { currentSeason } from '../dates';
+
+import type { BuilderInfo } from './interfaces';
+
+export async function getStarterpackBuilders(): Promise<BuilderInfo[]> {
```
Comment (mattcasey) [inline packages/scoutgame/src/builders/getStarterpackBuilders.ts:7]: ideally season is passed into the method and not imported globally

File: packages/scoutgame/src/builders/getTodaysHotBuilders.ts [hunk 11]
```diff
@@ -111,7 +118,9 @@ export async function getTodaysHotBuilders(): Promise<BuilderInfo[]> {
       rank: builder.userWeeklyStats[0]?.rank || -1,
       last7DaysGems: ((builder.builderCardActivities[0]?.last7Days as unknown as Last7DaysGems) || [])
         .map((gem) => gem.gemsCount)
-        .slice(-7)
+        .slice(-7),
+      contractAddress: builder.builderNfts[0]?.contractAddress || '',
```
Comment (mattcasey) [inline packages/scoutgame/src/builders/getTodaysHotBuilders.ts:122]: instead of passing an empty string to circumvent Typescript logic, contractAddress should probably be optional or u should throw an error

File: packages/scoutgame/src/constants.ts [hunk 12]
```diff
@@ -3,3 +3,8 @@ export const rewardPoints = 5;
 
 export const TELEGRAM_BOT_TOKEN = process.env.TELEGRAM_BOT_TOKEN;
 export const telegramBotName = 'ScoutGameXYZBot';
+
+/**
+ * Max amount of starter pack NFTs a user can mint
+ */
+export const MAX_STARTER_PACK_PURCHASES = 3;
```
Comment (mattcasey) [inline packages/scoutgame/src/constants.ts:10]: suggestion: put this next to the starter pack contract constants

File: packages/scoutgame/src/points/builderEvents/sendPointsForMiscEvent.ts [hunk 13]
```diff
@@ -80,3 +80,12 @@ export async function sendPointsForMiscEvent({
     return prisma.$transaction(txHandler);
   }
 }
+
+// sendPointsForMiscEvent({
```
Comment (mattcasey) [inline packages/scoutgame/src/points/builderEvents/sendPointsForMiscEvent.ts:84]: delete this and stop caling functions from code

File: packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+        ]),
+        pointsForBuilder: 480
+      })
+    );
+
+    const totalPointsDistributed = result.pointsPerScout.reduce((acc, scout) => acc + scout.scoutPoints, 0);
+    expect(totalPointsDistributed + result.pointsForBuilder).toBeLessThanOrEqual(result.earnableScoutPoints);
+  });
+
+  // it('should apply correct multipliers for NFT types', async () => {
```
Comment (mattcasey) [inline packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts:83]: these are commented out

File: packages/scoutgame/src/builderNfts/builderRegistration/createBuilderNftStarterPack.ts
Comment (mattcasey) [inline packages/scoutgame/src/builderNfts/builderRegistration/createBuilderNftStarterPack.ts:1]: builder is repeated 3 times in this path. just sayin'... :)

**Review by mattcasey — COMMENTED** [review]
Left some suggestions, with one top request:
- "starter_pack" should be a new contract name, because it is a different contract from the default one. I guess "prd_starter_pack". The concept of contractName sort of overlaps the nftType. You should never have the same contract name with a different nftType or vice versa. I think so long as we use one or the other depending on the context, it will be fine. But we shouldn't need to combine them for example in s3.
- Would also be nice to see some more tests, like an e2e that makes sure the starter pack appears

File: public/mockServiceWorker.js (hunk 2 above)
Comment (motechFR) [reply public/mockServiceWorker.js:11]: Checked file back to main state

File: packages/scoutgame/src/points/dividePointsBetweenBuilderAndScouts.ts (hunk 3 above)
Comment (motechFR) [reply packages/scoutgame/src/points/dividePointsBetweenBuilderAndScouts.ts:49]: Thanks, fixed this

File: packages/scoutgame/src/builderNfts/constants.ts (hunk 4 above)
Comment (motechFR) [reply packages/scoutgame/src/builderNfts/constants.ts:34]: Resorted to using this as I had undefined issues in the past

File: packages/scoutgame/src/builders/approveBuilderStarterPack.ts (hunk 6 above)
Comment (motechFR) [reply packages/scoutgame/src/builders/approveBuilderStarterPack.ts:45]: Fixed

File: apps/scoutgamecron/src/tasks/resolveMissingPurchases/index.ts (hunk 7 above)
Comment (motechFR) [reply apps/scoutgamecron/src/tasks/resolveMissingPurchases/index.ts:10]: Fixed to use nft type for now

File: packages/scoutgame/src/builders/getTodaysHotBuilders.ts (hunk 11 above)
Comment (motechFR) [reply packages/scoutgame/src/builders/getTodaysHotBuilders.ts:123]: Dropped contract address as it was part of iterations

File: packages/scoutgame-ui/src/components/common/Carousel/Carousel.tsx [hunk 15]
```diff
@@ -30,47 +33,64 @@ export function Carousel({ children, renderBullet, slidesPerView, height }: Caro
...
-        </Swiper>
-        {isDesktop && children.length > slidesPerView && (
-          <>
-            <NextArrow className='swiper-button-next' />
-            <PrevArrow className='swiper-button-prev' />
-          </>
-        )}
-      </Box>
+    <Box
+      width='90svw'
```
Comment (mattcasey) [inline packages/scoutgame-ui/src/components/common/Carousel/Carousel.tsx:46]: this looks like a pretty custom rule. margin-bottom is also very opinionated for a shared component. These changes messed up the carousel on Scout page

File: apps/scoutgame/app/(general)/scout/page.tsx [hunk 16]
```diff
@@ -13,6 +21,28 @@ export default async function Scout({
...
   const tab = (searchParams.tab as string) || 'scouts';
 
+  const user = await getUserFromSession();
+
+  let builders: BuilderInfo[] = [];
+
+  let remainingStarterCards = MAX_STARTER_PACK_PURCHASES;
+
+  if (user?.id) {
+    const purchases = await prisma.nFTPurchaseEvent
```
Comment (mattcasey) [inline apps/scoutgame/app/(general)/scout/page.tsx:31]: This query should be done in a child componennt so that it doesn't block the page from loading

File: apps/scoutgame/app/(general)/scout/page.tsx [hunk 17]
```diff
@@ -13,6 +21,28 @@ export default async function Scout({
...
+      .aggregate({
+        where: { builderNft: { nftType: 'starter_pack', season: currentSeason }, scoutId: user.id },
```
Comment (mattcasey) [inline apps/scoutgame/app/(general)/scout/page.tsx:33]: this should be stored on user stats

File: apps/scoutgame/app/(info)/welcome/page.tsx [hunk 18]
```diff
@@ -17,11 +17,6 @@ export const metadata: Metadata = {
 export default async function Welcome() {
   const user = await getUserFromSession();
 
-  if (user?.onboardedAt && user?.agreedToTermsAt && !user?.builderStatus) {
```
Comment (mattcasey) [inline apps/scoutgame/app/(info)/welcome/page.tsx:20]: This scenario has happened 19 times in the past 2 weeks. Are you sure we should remove it? https://app.datadoghq.com/logs?query=%22Redirect%20user%20to%20github%20connect%20page%20from%20Welcome%20page%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1732168220994&to_ts=1733464220994&live=true

File: packages/scoutgame/src/builderNfts/artwork/utils.ts (hunk 1 above)
Comment (mattcasey) [reply packages/scoutgame/src/builderNfts/artwork/utils.ts:25]: @motechFR don't forget to change this, please. so contractName is passed in and we use a different one for the starter-pack

File: packages/scoutgame/src/builderNfts/builderRegistration/createBuilderNft.ts [hunk 19]
```diff
@@ -22,6 +21,7 @@ export async function createBuilderNft({
   avatar: string | null;
   tokenId: bigint;
   builderId: string;
+  starterNft?: boolean;
```
Comment (mattcasey) [inline packages/scoutgame/src/builderNfts/builderRegistration/createBuilderNft.ts:24]: this isnt used

File: packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
+    starterPackNft = await mockBuilderNft({ builderId: builder.id, season, nftType: 'starter_pack' });
+
+    scout1 = await mockScout();
+    scout2 = await mockScout();
+
+    await mockNFTPurchaseEvent({
+      builderId: builder.id,
+      scoutId: scout1.id,
+      season,
+      tokensPurchased: 10
```
Comment (mattcasey) [inline packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts:33]: these purchase events are very specific to the test results about calculating starterpack NFTs and dont belong in a beforeAll().

Ideally, we would have at least 3-4 tests on the logic to calculate scout and builder points, each with their own scout and NFT situations

File: packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts [hunk 21]
```diff
@@ -0,0 +1,107 @@
...
+      season,
+      rank,
+      weeklyAllocatedPoints,
+      normalisationFactor
+    });
+
+    expect(result).toMatchObject(
+      expect.objectContaining({
+        // 10 * 10 (normal NFT) + 10 * 20 (normal NFT) +  1 * 10 (starter pack NFT) = 310
+        totalNftsPurchased: 310,
```
Comment (mattcasey) [inline packages/scoutgame/src/points/__tests__/dividePointsBetweenBuilderAndScouts.spec.ts:65]: we should rename this prop - it is not how many NFTs were purchased, that's confusing in a very critical part of our code
